resources/generate_book.py

```diff
@@ -475,10 +545,15 @@ def make_book(rm: ResourceManager, i18n: I18n, local_instance: bool = False):
             # todo: other just general heating recipes?
         )),
         entry('charcoal_forge', 'Charcoal Forge', 'tfc:textures/block/devices/charcoal_forge/lit_static.png', pages=(
-            text('Forging and stuff idk'),
-            multiblock('A Charcoal Forge', '', False, (('   ', ' 0 ', '   '), ('XXX', 'XCX', 'XXX'),), {'X': 'tfc:rock/bricks/andesite', 'C': 'tfc:charcoal_forge[heat_level=7]'}),
-            image('tfc:textures/gui/book/wip.png'),
-            # todo: chargoal forge
+            multiblock('A Charcoal Forge', 'A complete forge multiblock, ready to be lit.', True, multiblock_id='tfc:charcoal_forge'),
+            text('The $(thing)charcoal forge$() is a device used to $(l:mechanics/heating)heat$() and melt items. Forges are necessary to make a $(l:mechanics/crucible)crucible$() work. They are typically used to heat items to prepare them for $(l:mechanics/anvils)anvils$(). $(br)It is constructed with 5 $(item)stone$() blocks surrounding a $(thing)charcoal pile$() of 7 or 8 layers, which is then lit.'),
```

Comment:
Switch the order of these two pages - text page first allows there to be a proper title for this chapter.